stubs/paramiko/paramiko/win_pageant.pyi

```diff
@@ -3,12 +3,14 @@ import sys
 
 if sys.platform == "win32":
     win32con_WM_COPYDATA: int
-    def can_talk_to_agent(): ...
+    def can_talk_to_agent() -> bool: ...
+
+    ULONG_PTR: ctypes.c_uint64 | ctypes.c_uint32
 
     class COPYDATASTRUCT(ctypes.Structure): ...
 
     class PageantConnection:
         def __init__(self) -> None: ...
         def send(self, data: bytes) -> None: ...
-        def recv(self, n: int) -> bytes: ...
+        def recv(self, n: int) -> str | bytes: ...
```

Comment:
Presumably the best way to check the return value is `if foo == ""`, rather than `if isinstance(foo, str)`.

```suggestion
        def recv(self, n: int) -> Literal[""] | bytes: ...
```

You will also need to add an import.

Reply:
I wonder if their code at https://github.com/paramiko/paramiko/blob/1824a27c644132e5d46f2294c1e2fa131c523559/paramiko/win_pageant.py#L136 has a bug and should compare against `b""` instead.